File: src/frontend/lib/dto/doctorDto.dart
import 'scheduleDto.dart';

class DoctorDto {
  final String doctorId;
  final String name;
  final String email;
  final String gender;
  final double? averageRating;
  final String areaOfKnowledge;
  final String qualifications;
  final int yearsOfExperience;
  final String scfhsNumber;
  final String iban;
  final String? profilePhotoURL;
  final String? fcmToken;
  final String registrationStatus;
  final List<ScheduleDto>? schedule;

  DoctorDto({
    required this.doctorId,
    required this.name,
    required this.email,
    required this.gender,
    this.averageRating,
    required this.areaOfKnowledge,
    required this.qualifications,
    required this.yearsOfExperience,
    required this.scfhsNumber,
    required this.iban,
    this.profilePhotoURL,
    this.fcmToken,
    required this.registrationStatus,
    this.schedule,
  });

  Map<String, dynamic> toJson() {
    return {
      'doctorId': doctorId,
      'name': name,
      'email': email,
      'gender': gender,
      'averageRating': averageRating,
      'areaOfKnowledge': areaOfKnowledge,
      'qualifications': qualifications,
      'yearsOfExperience': yearsOfExperience,
      'scfhsNumber': scfhsNumber,
      'iban': iban,
      'profilePhotoURL': profilePhotoURL,
      'fcmToken': fcmToken,
      'registrationStatus': registrationStatus,
      'schedule': schedule?.map((s) => s.toJson()).toList(),
    };
  }
}

File: src/backend/src/main/java/com/bouh/backend/model/Dto/doctorDto.java
package com.bouh.backend.model.Dto;
import com.google.cloud.firestore.annotation.DocumentId;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;


@Data //setters,getters
@NoArgsConstructor
@AllArgsConstructor
public class doctorDto {
    @DocumentId
    private String doctorId;

    private String name;
    private String email;
    private String gender;
    private Double averageRating;
    private String areaOfKnowledge;
    private List<String> qualifications;
    private Integer yearsOfExperience;
    private String scfhsNumber;
    private String iban;
    private String profilePhotoURL;
    private String fcmToken;
    private String registrationStatus;
    }
